test/common/http/http2/http2_connection_fuzz_test.cc

```diff
@@ -0,0 +1,307 @@
+#include <alloca.h>
+
+#include <string>
+#include <vector>
+
+#include "source/common/http/http2/codec_impl.h"
+
+#include "test/common/http/http2/http2_connection.pb.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/test_common/test_runtime.h"
+
+#include "absl/strings/string_view.h"
+#include "nghttp2/nghttp2.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http2 {
+namespace {
+
+static const uint32_t MAX_HEADERS = 32;
+static const size_t MAX_HD_TABLE_BUF_SIZE = 4096;
+
+class Http2Frame {
```

Comment:
My only suggestion would to be use this class if possible: https://github.com/envoyproxy/envoy/blob/beecb4c563edabbd61b867891b54ab4888be7a5d/test/common/http/http2/http2_frame.h#L31

Maybe move the functionality that you need into the existing class.

Reply:
Done.